src/service/app_state.py

```diff
@@ -46,9 +56,33 @@ async def build_app(app: FastAPI) -> None:
     auth = await KBaseAuth.create(auth_url, full_admin_roles=admin_roles)
     logger.info("KBase auth service connected")
 
+    # Initialize MinIO configuration and client
+    logger.info("Initializing MinIO client and managers...")
+    config = MinIOConfig(
+        endpoint=not_falsy(os.getenv("MINIO_ENDPOINT"), "MINIO_ENDPOINT"),
+        access_key=not_falsy(os.getenv("MINIO_ROOT_USER"), "MINIO_ROOT_USER"),
+        secret_key=not_falsy(os.getenv("MINIO_ROOT_PASSWORD"), "MINIO_ROOT_PASSWORD"),
+    )
+
+    minio_client = MinIOClient(config)
+    await minio_client.initialize_session()
+    logger.info("MinIO client session initialized")
+
+    # Initialize all managers with the shared client
+    user_manager = UserManager(minio_client, config)
+    group_manager = GroupManager(minio_client, config)
+    policy_manager = PolicyManager(minio_client, config)
```

Comment:
Can any of these throw an exception ? if so, should try/finally close the minio client

Reply:
:+1: